core/algorithms/onpolicy_sync/engine.py

```diff
@@ -624,13 +624,18 @@ def __init__(
         self.last_log: Optional[int] = None
         self.last_save: Optional[int] = None
 
-    def advance_seed(self, seed: Optional[int]) -> Optional[int]:
+    def advance_seed(
+        self, seed: Optional[int], call_worker_seeds=True
+    ) -> Optional[int]:
         if seed is None:
             return seed
         seed = (seed ^ (self.training_pipeline.total_steps + 1)) % (
```

Comment:
Shouldn't this be `(seed ** (self.training_pipeline.total_steps + 1)) % (2 ** 31 - 1)`. The `^` is a bitwise operation and is pretty strange. If so we should have it be `pow(seed, self.training_pipeline.total_steps + 1, 2 ** 31 - 1)` as this will make things more efficient.

Reply:
We've discussed this already a couple of times. I actually designed it as a bitwise operation and checked that the seeds were not too correlated and things looked fine. I'm reluctant to change something that works. Bitwise XORs are pretty common in RNG internals...

Reply:
It just surprises me anew every time :) Fair enough, can you add a comment for my sanity saying explicitly that this is meant to be a bitwise operation so I don't make the comment in a few months again?

Reply:
I'll add this.